module/zfs/zfs_rlock.c

```diff
@@ -106,10 +106,15 @@ zfs_range_lock_writer(znode_t *zp, rl_t *new)
 	avl_tree_t *tree = &zp->z_range_avl;
 	rl_t *rl;
 	avl_index_t where;
-	uint64_t end_size;
 	uint64_t off = new->r_off;
 	uint64_t len = new->r_len;
 
+#ifdef _KERNEL
+	uint64_t end_size;
+#else
+	ASSERT3U(zp->z_is_zvol, ==, B_TRUE);
+#endif
```

Comment:
I believe z_is_zvol is used by the zfs_range_lock_writer() to decide if this is a 'simple' (for zvols) or 'compicated' (for real znodes - files) implementation. This member simplifies things quite a bit in terms of using 'user space znode structure.